### ISSUE TITLE
Deleting a reward asks you if you want to delete the *task*

### ISSUE THREAD

**Issue by sgabler** [HabitRPG/habitica#6146]
![image](https://cloud.githubusercontent.com/assets/3625477/10850649/1c2fe104-7f27-11e5-9eb3-c96c1d14a0f6.png)

The text should read "Are you sure you want to delete this reward?"

_EDIT by Alys 2019-08-29:_
See https://github.com/HabitRPG/habitica/issues/6146#issuecomment-523241278 for the current status of this bug and the desired fix.

**Comment by Thrashmandicoot** [comment]
confirmed.

line 89 tasks.json is "Are you sure you want to delete this task?"

Simple recommended fix of changing to "Are you sure you want to delete this?" since used commonly for rewards and tasks.

Other fix could be taking a json value that asks if they want to delete this 'daily', 'to-do', 'reward', etc.

**Comment by Alys** [comment]
+1 for "Are you sure you want to delete this?"

I would love it to also state the text of the item you're deleting. That would protect the user from accidentally clicking on the wrong trash can icon.

**Comment by Alys** [comment]
I'm reopening this because it's happening again. 

I'm assuming that the desired fix is the [same one that was implemented previously](https://github.com/HabitRPG/habitica/pull/6314) but I'll leave this as suggestion-discussion for a few days in case there's other preferences.

If the same fix is done, the text shown in that PR should be adjusted to use the correct form for the task types ("Habit", "Daily", "To-Do", "Reward"), or it should use the strings that already exist for that (e.g., `tasks.json:  "todo": "To-Do"`).

**Comment by bmasetto** [comment]
@Alys, I agree to the suggestion about having diferent texts to each task type. This approach gives the user the entire corcerning about about its action. Can I be assigned to this issue? I wanna to work on this issue :)

**Comment by Alys** [comment]
@bmasetto Thanks! Please go ahead. I've marked it as in progress for you.

**Comment by thatrobotdev** [comment]
Can I take this one?

**Comment by bmasetto** [comment]
In my opinion, you can, @thatrobotdev! Unfortunately, I was not able to start this issue development. @Alys, is it okay for you if @thatrobotdev take this issue?

**Comment by Alys** [comment]
That's fine! Thanks @bmasetto  and @thatrobotdev !

**Comment by thatrobotdev** [comment]
Awesome! I'll start work on it as soon as possible, but I'll definitely be able to focus on it this weekend! :D

**Comment by thatrobotdev** [comment]
Sorry, I am not able to successfully install Habitica on my local system. I can't work on this issue anymore.

**Comment by Alys** [comment]
@thatrobotdev I'm sorry about that and thank you for telling us. I've marked this as help wanted. 

If you'd like help with your local install, tell us what's going wrong. If you've already posted to the Aspiring Blacksmiths guild about it and received no replies, please post a reminder.

**Comment by thatrobotdev** [comment]
I might be able to help on Habitica in the future, if I can devote more time to troubleshooting this! ;) Thanks for your help.

**Comment by zack25** [comment]
Hello, I'd like to work on this.

**Comment by Alys** [comment]
@zack25 Thanks very much! Please go ahead. Post here if you have questions.

**Comment by zack25** [comment]
I noticed that when editing a Task that the link to delete it reads: "Delete this Task".
![image](https://user-images.githubusercontent.com/19578293/69491445-eb8db680-0e5a-11ea-9dde-584645c527cb.png)

Should I also change this so that it states "Delete this [Task Type]"? This ticket only mentions changing the confirmation message, but I think it'd be consistent to make this change as well, unless there are any objections?
![image](https://user-images.githubusercontent.com/19578293/69491452-0ceea280-0e5b-11ea-8009-f7faf52f77a9.png)

**Comment by Alys** [comment]
@zack25 Good catch! I think you're right that that should be changed for all four "task" types. At the very least the current text is wrong for Rewards and should be fixed, and changing it for the other three types for consistency seems good to me.

**Comment by shanaqui** [comment]
@zack25 Hi! Are you still interested in working on this? Please let us know. :)

**Comment by shanaqui** [comment]
I'll mark this as help wanted again since there was no reply!

**Comment by zack25** [comment]
Sorry, my computer crashed, and then we got busy moving to a new place. I
won't be able to work on this for a while.

**Comment by Shuyinsama** [comment]
I would love to take this one. It seems we only need to add the correct translations and change both the window alert message as well as the text in the edit task modal?

**Comment by Alys** [comment]
@Shuyinsama Thanks! I've marked this as in progress for you.

You don't need to worry about translations at all (see the _README_FIRST.md file in any of the locales directories for more details if you're curious).

> change both the window alert message as well as the text in the edit task modal?

Yes, that sounds correct. I've deleted some old comments that don't directly relate to the fix required, so if you'd like to look through the issue again it should be easier to find the latest requirements. :) See in particular this comment: https://github.com/HabitRPG/habitica/issues/6146#issuecomment-523241278 and the couple just after it.

Post again here if you have any other questions or if anything else needs clarification!
